bridge/github/export.go

```
@@ -0,0 +1,654 @@
+			panic("expected to find github issue URL")
+		}
+
+		// will be used to mark operation related to a bug as exported
+		bugGithubID = githubID
+		bugGithubURL = githubURL
+
+	} else {
+		// check that we have a token for operation author
+		client, err := ge.getIdentityClient(author.Id())
+		if err != nil {
+			// if bug is still not exported and user cannot author bug stop the execution
+
+			// TODO: maybe print a warning ?
+			// this is not an error
+			// don't export bug
+			return nil
+		}
+
+		// create bug
+		id, url, err := createGithubIssue(client, ge.repositoryID, createOp.Title, createOp.Message)
+		if err != nil {
+			return fmt.Errorf("creating exporting github issue %v", err)
+		}
+
+		hash, err := createOp.Hash()
+		if err != nil {
+			return fmt.Errorf("comment hash: %v", err)
+		}
+
+		// mark bug creation operation as exported
+		if err := markOperationAsExported(b, hash, id, url); err != nil {
+			return fmt.Errorf("marking operation as exported: %v", err)
+		}
+
+		// cache bug github ID and URL
+		bugGithubID = id
+		bugGithubURL = url
+	}
+
+	// get createOp hash
+	hash, err := createOp.Hash()
+	if err != nil {
+		return err
+	}
+
+	bugCreationHash = hash.String()
+
+	// cache operation github id
+	ge.cachedIDs[bugCreationHash] = bugGithubID
+
+	for _, op := range snapshot.Operations[1:] {
+		// ignore SetMetadata operations
+		if _, ok := op.(*bug.SetMetadataOperation); ok {
+			continue
+		}
+
+		// get operation hash
+		hash, err := op.Hash()
+		if err != nil {
+			return fmt.Errorf("reading operation hash: %v", err)
+		}
+
+		// ignore imported (or exported) operations from github
+		// cache the ID of already exported or imported issues and events from Github
+		if id, ok := op.GetMetadata(keyGithubId); ok {
+			ge.cachedIDs[hash.String()] = id
+			continue
+		}
+
+		opAuthor := op.GetAuthor()
+		client, err := ge.getIdentityClient(opAuthor.Id())
+		if err != nil {
+			// don't export operation
+			continue
```

maybe use a specific error for "missing token" and test it here ? That way just this case is ignored and a legitimate error would not be silently dropped

---

Done !